docs/NNNN-hash-and-id-n-tuple-storage-layout.md

```diff
@@ -0,0 +1,323 @@
+
+## Procedure
+
+The following is an outline of the steps to follow to map an OCFL object identifier to an OCFL object root path using this extension (also see the Python Code section):
+
+1. The OCFL object identifier is encoded as UTF-8 and hashed using the specified `digestAlgorithm`.
+2. The digest is encoded as a lower-case hex string.
+3. Starting at the beginning of the digest and working forwards, the digest is divided into `numberOfTuples` tuples each containing `tupleSize` characters.
+4. The tuples are joined, in order, using the filesystem path separator.
+5. The OCFL object identifier is percent-encoded to create the encapsulation directory name (see ["Encapsulation Directory"](#encapsulation-directory) section above for details).
+6. The encapsulation directory name is joined to the end of the path.
+
+## Examples
+
+### Example 1
+
+This example demonstrates what the OCFL storage hierarchy looks like when using this extension's default configuration.
+
+#### Parameters
+
+It is not necessary to specify any parameters to use the default configuration. However, if you were to do so, it would look like the following:
+
+```json
+{
+    "extensionName": "NNNN-hash-and-id-n-tuple-storage-layout",
+    "digestAlgorithm": "sha256",
+    "tupleSize": 3,
+    "numberOfTuples": 3
+}
+```
+
+#### Mappings
+
+| Object ID | Digest | Object Root Path |
+| --- | --- | --- |
+| object-01 | 3c0ff4240c1e116dba14c7627f2319b58aa3d77606d0d90dfc6161608ac987d4 | `3c0/ff4/240/object-01` |
+| ..hor/rib:le-$id | 487326d8c2a3c0b885e23da1469b4d6671fd4e76978924b4443e9e3c316cda6d | `487/326/d8c/%2e%2ehor%2frib%3ale-%24id` |
+
+#### Storage Hierarchy
+
+```
+[storage_root]/
+├── 0=ocfl_1.0
+├── ocfl_layout.json
+├── NNNN-hashed-n-tuple-id-layout.json
+├── 3c0/
+│   └── ff4/
+│       └── 240/
+│           └── object-01/
+│               ├── 0=ocfl_object_1.0
+│               ├── inventory.json
+│               ├── inventory.json.sha512
+│               └── v1 [...]
+└── 487/
+    └── 326/
+        └── d8c/
+            └── %2e%2ehor%2frib%3ale-%24id/
+                ├── 0=ocfl_object_1.0
+                ├── inventory.json
+                ├── inventory.json.sha512
+                └── v1 [...]
+```
+
+### Example 2
+
+This example demonstrates the effects of modifying the default parameters to use a different `digestAlgorithm`, smaller `tupleSize`, and a larger `numberOfTuples`.
+
+#### Parameters
+
+```json
+{
+    "digestAlgorithm": "md5",
+    "tupleSize": 2,
+    "numberOfTuples": 15
+}
+```
+
+#### Mappings
+
+| Object ID | Digest | Object Root Path |
+| --- | --- | --- |
+| object-01 | ff75534492485eabb39f86356728884e | `ff/75/53/44/92/48/5e/ab/b3/9f/86/35/67/28/88/object-01` |
+| ..hor/rib:le-$id | 08319766fb6c2935dd175b94267717e0 | `08/31/97/66/fb/6c/29/35/dd/17/5b/94/26/77/17/%2e%2ehor%2frib%3ale-%24id` |
+
+#### Storage Hierarchy
+
+```
+[storage_root]/
+├── 0=ocfl_1.0
+├── ocfl_layout.json
+├── NNNN-hashed-n-tuple-id-layout.json
```

Comment:
This file should now be at `extensions/hash-and-id-n-tuple-storage-layout/config.json`